qiskit/circuit/quantumcircuit.py

```diff
@@ -607,21 +609,21 @@ def qubits(self):
         """
         Returns a list of quantum bits in the order that the registers were added.
         """
-        return [qbit for qreg in self.qregs for qbit in qreg]
+        return self._qubits
 
     @property
     def clbits(self):
         """
         Returns a list of classical bits in the order that the registers were added.
         """
-        return [cbit for creg in self.cregs for cbit in creg]
+        return self._clbits
 
     @property
     def ancillas(self):
         """
         Returns a list of quantum bits in the order that the registers were added.
         """
-        return [qbit for qreg in self.qregs for qbit in qreg if isinstance(qbit, AncillaQubit)]
+        return [qbit for qbit in self._qubits if isinstance(qbit, AncillaQubit)]
```

Comment:
can't ancillas be cached in the same way? I don't think qubit types change in a circuit. Also the docstring here seems like it needs to be updated.

Reply:
Added in 04eeb6a29 . Updated the docstring in a581d30 .